Content 2 6 release (#2293)
FlowFuse/website: src/blog/2024/07/flowfuse-2-6-release.md

---
title: "FlowFuse 2.6: AI Infused Node-RED, Persistent File Storage & Lots More"
subtitle: Lowering the barrier to entry for new users, and enhancing the flexibility and functionality of the platform.
description: Discover the new features in FlowFuse 2.6, and it's focus on improving the Node-RED development experience.
date: 2024-07-04
authors: ["joe-pavitt"]
image: /blog/2024/07/images/release-2-6-july-2024.png
tags:
   - posts
   - flowfuse
   - releases

>    - LDAP
>    - snapshot
>    - blueprint

Reviewer: ```suggestion
   - storage
   - editor
   - ai
   - assistant
   - blueprint
```

![Screenshot showing the "FlowFuse Assistant" button available in the Editor Toolbar](./images/assistant-toolbar.png){data-zoomable}
_Screenshot showing the "FlowFuse Assistant" button available in the Editor Toolbar_

![Screenshot showing the "Ask the FlowFuse Assistant" button available in the function node](./images/assistant-function-node-inline-code-lens.png){data-zoomable}
_Screenshot showing the "Ask the FlowFuse Assistant" button available in the function node_

The Node-RED Assistant is an AI-powered tool that can help you write JavaScript code in the function node. It can suggest code snippets, help you debug your code, and even write code for you. This is a game-changer for users who are inexperienced with JavaScript, as it lowers the barrier to entry and makes it easier to create applications.

### Immersive Editor Experience

> When we were investigating how users are using FlowFuse and Node-RED together, we found a lot of users having to jump back and forth between the Node-RED Editor and FlowFuse UI, to view logs, save snapshots, restart the Editor after updates, etc.

Reviewer 2: ```suggestion
During our investigation on how FlowFuse and Node-RED are used together, we found out that many users have to frequently switch between the Node-RED Editor and FlowFuse UI. This back-and-forth movement is often necessary to view logs, save snapshots, restart the Editor after updates, and perform other tasks.
```

We're always seeking to reduce friction in the FlowFuse user experience, and as such, we've introduced a large overhaul of the developer experience for Node-RED when running in FlowFuse, in what we're calling the "Immersive Editor".

![Screenshot showing the "Immersive Editor" view](./images/immersive-editor.png){data-zoomable}
_Screenshot showing the "Immersive Editor" view, where the FlowFuse navigation is available as a floating bar._

Now, the instances tabs are all available in the _same_ view as the Editor.

![Screenshot showing the "Collapsed" FlowFuse bar](./images/immersive-editor-collapsed.png){data-zoomable}
_Screenshot showing the "FlowFuse" button at the bottom of the Node-RED Editor._

Don't worry though, if you want the full editor experience again, you can just collapse the FlowFuse menus down to a little "FlowFuse" button at the bottom.

> ## Persistent File Storage
> 
> This now unlocks many Node-RED nodes that previously couldn't run in the FlowFuse Cloud environment. Just a few now available are:
> 
> - `node-red-node-sqlite` - A Node-RED node to read and write to a SQLite database.
> - ADD OTHERS HERE
> 
> // TODO: THIS NEEDS MORE CONTENT
> 

Reviewer: ```suggestion
## Persistent File Storage

Since the early days of FlowFuse, we have provided custom File nodes that can be used to read and write individual files from a flow. This was necessary because the local file system was not considered persistent; restarting an instance would reset the file system back to how it was when the instance first started. 

Whilst this solved the immediate problem for the File nodes, we know there were 3rd party nodes that would want to use the file system as well - and we couldn't expect them to update to work with our custom solution.

With the 2.6 release, each instance now gets a piece of persistent file system they can read and write to normally, from any node - with full confidence those files will be persisted between restarts.

This unlocks lots of new capabilities using nodes from the community. For example, the [SQLite](https://flows.nodered.org/node/node-red-node-sqlite) nodes can be used to quickly add a locally managed database to store your data in.

All newly created instances of FlowFuse Cloud from today will have this storage enabled. If you have an existing instance you'd like to move over, then do get [in touch](https://flowfuse.com/contact-us/) and we can help move you over.

```

![Screenshot showing the "FlowFuse Assistant" button available in the Editor Toolbar](./images/assistant-toolbar.png){data-zoomable}
_Screenshot showing the "FlowFuse Assistant" button available in the Editor Toolbar_

![Screenshot showing the "Ask the FlowFuse Assistant" button available in the function node](./images/assistant-function-node-inline-code-lens.png){data-zoomable}
_Screenshot showing the "Ask the FlowFuse Assistant" button available in the function node_

The Node-RED Assistant is an AI-powered tool that can help you write JavaScript code in the function node. It can suggest code snippets, help you debug your code, and even write code for you. This is a game-changer for users who are inexperienced with JavaScript, as it lowers the barrier to entry and makes it easier to create applications.

### Immersive Editor Experience

> During our investigation on how FlowFuse and Node-RED are used together, we found out that many users have to frequently switch between the Node-RED Editor and FlowFuse UI. This back-and-forth movement is often necessary to view logs, save snapshots, restart the Editor after updates, and perform other tasks.

Author: ```suggestion
During our investigation on how FlowFuse and Node-RED are used together, we found out that many users have to frequently switch between the Node-RED Editor and FlowFuse UI. This back-and-forth movement was often necessary to view logs, save snapshots, restart the Editor after updates, and perform other tasks.
```

We're always seeking to reduce friction in the FlowFuse user experience, and as such, we've introduced a large overhaul of the developer experience for Node-RED when running in FlowFuse, in what we're calling the "Immersive Editor".

![Screenshot showing the "Immersive Editor" view](./images/immersive-editor.png){data-zoomable}
_Screenshot showing the "Immersive Editor" view, where the FlowFuse navigation is available as a floating bar._

Now, the instances tabs are all available in the _same_ view as the Editor.

![Screenshot showing the "Collapsed" FlowFuse bar](./images/immersive-editor-collapsed.png){data-zoomable}
_Screenshot showing the "FlowFuse" button at the bottom of the Node-RED Editor._

Don't worry though, if you want the full editor experience again, you can just collapse the FlowFuse menus down to a little "FlowFuse" button at the bottom.

> 

Reviewer: ```suggestion

The new Immersive Editor is available for instances running Node-RED 4.0.2 or later - older versions of Node-RED will still use the separate views.

```
